---
title: "Thieringer et al. Edgar Mine Workflow for 16S Data and Geochemistry"
output: html_notebook
---

# 16S Amplicon Analysis

#### Saved text of sessionInfo() as well as the packages used in the submission of this manuscript

R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggpubr_0.4.0                lubridate_1.7.9.2           ggthemr_1.1.0               scales_1.1.1                QsRutils_0.1.4             
 [6] vegan_2.5-7                 lattice_0.20-41             permute_0.9-5               ampvis2_2.6.5               phyloseqGraphTest_0.1.0    
[11] phylosmith_1.0.5            devtools_2.3.2              usethis_2.0.0               DESeq2_1.28.1               SummarizedExperiment_1.18.2
[16] DelayedArray_0.14.1         matrixStats_0.57.0          Biobase_2.50.0              GenomicRanges_1.40.0        GenomeInfoDb_1.24.2        
[21] knitr_1.30                  phangorn_2.5.5              ape_5.5                     DECIPHER_2.16.1             RSQLite_2.2.2              
[26] Biostrings_2.58.0           XVector_0.30.0              IRanges_2.24.1              S4Vectors_0.28.1            BiocGenerics_0.36.0        
[31] phyloseq_1.34.0             dada2_1.16.0                Rcpp_1.0.6                  gridExtra_2.3               ggplot2_3.3.3              
[36] BiocManager_1.30.10        

loaded via a namespace (and not attached):
  [1] utf8_1.2.1               tidyselect_1.1.1         AnnotationDbi_1.50.3     htmlwidgets_1.5.3        grid_4.0.2              
  [6] BiocParallel_1.22.0      Rtsne_0.15               munsell_0.5.0            codetools_0.2-18         units_0.6-7             
 [11] withr_2.4.2              colorspace_2.0-1         rstudioapi_0.13          ggsignif_0.6.0           labeling_0.4.2          
 [16] GenomeInfoDbData_1.2.3   hwriter_1.3.2            polyclip_1.10-0          bit64_4.0.5              farver_2.1.0            
 [21] rhdf5_2.34.0             rprojroot_2.0.2          vctrs_0.3.8              generics_0.1.0           xfun_0.23               
 [26] R6_2.5.0                 doParallel_1.0.16        graphlayouts_0.7.1       locfit_1.5-9.4           bitops_1.0-7            
 [31] rhdf5filters_1.2.0       assertthat_0.2.1         ggraph_2.0.4             gtable_0.3.0             multcompView_0.1-8      
 [36] processx_3.5.2           tidygraph_1.2.0          rlang_0.4.11             genefilter_1.70.0        splines_4.0.2           
 [41] rstatix_0.6.0            lazyeval_0.2.2           broom_0.7.3              abind_1.4-5              yaml_2.2.1              
 [46] reshape2_1.4.4           backports_1.2.1          ggnetwork_0.5.8          tools_4.0.2              ellipsis_0.3.2          
 [51] biomformat_1.16.0        RColorBrewer_1.1-2       sessioninfo_1.1.1        ggnet_0.1.0              plyr_1.8.6              
 [56] progress_1.2.2           zlibbioc_1.36.0          classInt_0.4-3           purrr_0.3.4              RCurl_1.98-1.2          
 [61] ps_1.6.0                 prettyunits_1.1.1        viridis_0.6.1            haven_2.3.1              ggrepel_0.9.0           
 [66] cluster_2.1.0            fs_1.5.0                 tinytex_0.28             magrittr_2.0.1           data.table_1.14.0       
 [71] openxlsx_4.2.3           pkgload_1.1.0            hms_1.1.0                patchwork_1.1.1          evaluate_0.14           
 [76] xtable_1.8-4             XML_3.99-0.5             rio_0.5.16               jpeg_0.1-8.1             readxl_1.3.1            
 [81] testthat_3.0.1           compiler_4.0.2           tibble_3.1.2             KernSmooth_2.23-18       crayon_1.4.1            
 [86] htmltools_0.5.1.1        mgcv_1.8-33              tidyr_1.1.3              geneplotter_1.66.0       RcppParallel_5.0.2      
 [91] DBI_1.1.0                tweenr_1.0.1             MASS_7.3-53              sf_0.9-7                 car_3.0-10              
 [96] ShortRead_1.46.0         Matrix_1.3-2             ade4_1.7-16              cli_2.5.0                quadprog_1.5-8          
[101] igraph_1.2.6             forcats_0.5.0            pkgconfig_2.0.3          GenomicAlignments_1.24.0 foreign_0.8-81          
[106] plotly_4.9.3             foreach_1.5.1            annotate_1.66.0          multtest_2.46.0          stringr_1.4.0           
[111] callr_3.7.0              digest_0.6.27            rmarkdown_2.6            cellranger_1.1.0         fastmatch_1.1-0         
[116] curl_4.3.1               Rsamtools_2.4.0          lifecycle_1.0.0          nlme_3.1-151             jsonlite_1.7.2          
[121] Rhdf5lib_1.12.0          carData_3.0-4            network_1.16.1           desc_1.2.0               viridisLite_0.4.0       
[126] fansi_0.5.0              pillar_1.6.1             httr_1.4.2               pkgbuild_1.2.0           survival_3.2-7          
[131] glue_1.4.2               remotes_2.2.0            zip_2.1.1                png_0.1-7                iterators_1.0.13        
[136] bit_4.0.4                ggforce_0.3.2            class_7.3-17             stringi_1.6.2            blob_1.2.1              
[141] latticeExtra_0.6-29      memoise_1.1.0            dplyr_1.0.6              e1071_1.7-4 

#### Reading in the phyloseq objects to use downstream
```{r}
em_ps <- readRDS("em_phyloseq_object.rds")
```

#### Removing sample EM3004 because there were no sequences that were generated/merged after creating the phyloseq object
```{r}
em_ps <- subset_samples(em_ps, sample_names(em_ps) != "EM3004")
```

#### Rarefaction of the original phyloseq object to the next lowest sample in order to include as many samples as possible while maintaining enough reads
```{r}
em_ps_rare <- rarefy_even_depth(em_ps, sample.size = 1270, rngseed = 700) #A sample size of 5,761 is used when looking only at borehole samples. The current sample size includes the soil samples observed in this study and to produce the PCOA and NMDS plots.
```


### PCOA Ordination Figure Creation
```{r}
ord.pcoa.wunifrac <- ordinate(em_ps_rare, method = "PCoA", distance = "unifrac", weighted = TRUE)
PCOA_plot <- plot_ordination(em_ps_rare, ord.pcoa.wunifrac, type = "samples", color = "Site_Collected_From", label = NULL) +
  geom_point(size = 6) +
  stat_ellipse(type = "norm") +
  labs(color = "Sample Source") +
  geom_point(size = 3) +
  scale_color_manual(values = c("#0072B2", "#F0E442", "#E69F00","#009E73", "#D55E00", "#CC79A7", "#000000"))
  #ggthemr('flat') #Add this at the beginning of this code chunk to match the aesthetics of the plot in the manuscript. To reset the palette for the additional figures use the function ggthemr_reset()
PCOA_plot
```


#### Removing addtional samples that contain too few reads to produce accurately representative borehole fluid samples, and using Hellinger transformation in order to observe NMDS ordination
```{r}
em_ps_subset <- subset_samples(em_ps, sample_names(em_ps) != "EM2006")
em_ps_subset <- subset_samples(em_ps_subset, sample_names(em_ps_subset) != "EM2011")
em_ps_subset <- subset_samples(em_ps_subset, sample_names(em_ps_subset) != "EM2009")
```

```{r}
ps_prop <- transform_sample_counts(em_ps_subset, function(otu) otu/sum(otu))
ps_prop_hell <- vegan_stand(ps_prop, method = "hellinger")
```


### NMDS Ordination Supplemental Figure Creation
```{r}
ord.nmds.bray <- ordinate(ps_prop_hell, method="NMDS", distance="bray")
em_nmds_days_since_previous <- plot_ordination(ps_prop, ord.nmds.bray, type = "samples", color= "Days_Since_Previous_Extraction", shape = "Site_Collected_From", title="Bray NMDS") +
  geom_point(size = 5) +
  labs(shape = "Sample Source", color = "Days Since Previous Extraction") +
  scale_shape_manual(values = c(15, 16, 17, 18, 23, 24, 25)) +
  theme_bw()
em_nmds_days_since_previous
```


#### Adonis Test
```{r}
edgar_bray <- phyloseq::distance(em_ps_rare, method = "bray")
edgardf <- data.frame(sample_data(em_ps_rare))

adonis(edgar_bray ~ Site_Collected_From, data = edgardf) #Can change the variable of interest to one observed in the study
```

### Rarefaction Curves
```{r}
em_otu <- otu_table(em_ps)

rarecurve(em_otu)
```


#### Formatting data to fit into Ampvis2 for heatmap visualization
```{r}
em_ps_boreholes_6_and_8 <- rarefy_even_depth(em_ps, sample.size = 5671, rngseed = 700) #rarefying to a deeper level to include all Borehole 6 + 8 samples
#Removing all samples belonging to Boreholes 2 + 3
em_ps_boreholes_6_and_8 <- subset_samples(em_ps_boreholes_6_and_8, sample_names(em_ps_boreholes_6_and_8) != "EM2004")
em_ps_boreholes_6_and_8 <- subset_samples(em_ps_boreholes_6_and_8, sample_names(em_ps_boreholes_6_and_8) != "EM2005")
em_ps_boreholes_6_and_8 <- subset_samples(em_ps_boreholes_6_and_8, sample_names(em_ps_boreholes_6_and_8) != "EM2006")
em_ps_boreholes_6_and_8 <- subset_samples(em_ps_boreholes_6_and_8, sample_names(em_ps_boreholes_6_and_8) != "EM2007")
em_ps_boreholes_6_and_8 <- subset_samples(em_ps_boreholes_6_and_8, sample_names(em_ps_boreholes_6_and_8) != "EM2008")
em_ps_boreholes_6_and_8 <- subset_samples(em_ps_boreholes_6_and_8, sample_names(em_ps_boreholes_6_and_8) != "EM2009")
em_ps_boreholes_6_and_8 <- subset_samples(em_ps_boreholes_6_and_8, sample_names(em_ps_boreholes_6_and_8) != "EM2010")
em_ps_boreholes_6_and_8 <- subset_samples(em_ps_boreholes_6_and_8, sample_names(em_ps_boreholes_6_and_8) != "EM2011")
em_ps_boreholes_6_and_8 <- subset_samples(em_ps_boreholes_6_and_8, sample_names(em_ps_boreholes_6_and_8) != "EM2012")
em_ps_boreholes_6_and_8 <- subset_samples(em_ps_boreholes_6_and_8, sample_names(em_ps_boreholes_6_and_8) != "EM3001")
em_ps_boreholes_6_and_8 <- subset_samples(em_ps_boreholes_6_and_8, sample_names(em_ps_boreholes_6_and_8) != "EM3002")
em_ps_boreholes_6_and_8 <- subset_samples(em_ps_boreholes_6_and_8, sample_names(em_ps_boreholes_6_and_8) != "EM3003")
em_ps_boreholes_6_and_8 <- subset_samples(em_ps_boreholes_6_and_8, sample_names(em_ps_boreholes_6_and_8) != "EM3004")
em_ps_boreholes_6_and_8 <- subset_samples(em_ps_boreholes_6_and_8, sample_names(em_ps_boreholes_6_and_8) != "EM3005")
em_ps_boreholes_6_and_8 <- subset_samples(em_ps_boreholes_6_and_8, sample_names(em_ps_boreholes_6_and_8) != "EM3006")
sample_sums(em_ps_boreholes_6_and_8) #There should be a total of 6 samples
```

```{r}
obj <- em_ps_boreholes_6_and_8
t_otu <- t(data.frame(otu_table(obj)))
myotutable <- data.frame(OTU = rownames(t_otu@.Data),
                       t_otu@.Data,
                       phyloseq::tax_table(obj)@.Data,
                       check.names = FALSE
                       )
myotutable$Species <- NA
mymetadata <- data.frame(phyloseq::sample_data(obj), 
                       check.names = FALSE
                       )
em_ampvis2_obj <- amp_load(otutable = myotutable, metadata = mymetadata)
```


### Heatmap Visualization and Figure Creation
```{r}
em_amp_heatmap <- amp_heatmap(em_ampvis2_obj,
            group_by = "Collection_date",
            facet_by = "Site_Collected_From",
            tax_aggregate = "Order",
            tax_add = c("Phylum", "Class"),
            tax_empty = "remove",
            tax_show = 15,
            showRemainingTaxa = TRUE,
            plot_colorscale = "sqrt",
            function_data = TRUE,
            plot_values = TRUE) +
              theme(axis.text.x = element_text(angle = 0, size = 10, hjust = 0.5),
                    axis.text.y = element_text(size = 12),
                    legend.position = "right",
                    legend.text = element_text(size = 10),
                    legend.title = element_text(size = 10),
                    strip.text.x = element_text(size = 12, face = "bold"),
                    strip.background = element_rect(color = "black", size = 1.5, linetype = "solid"),
                    panel.border = element_rect(color = "grey", fill = NA, size = 1)) 
em_amp_heatmap
```


#### Importing data to use for abundance line figure creation. 
The ASV table contains all of the Phylum-level taxa present in the samples, where those below 5% abundance where summed into the "other" category. The full ASV table can be observed from the ASV table used for the phyloseq object.
```{r}
em_abundance_borehole2 <- read.csv("AbundanceLinesB2.csv")
em_abundance_borehole2$dates <- as.Date(em_abundance_borehole2$dates) #Need to do this in order for the figure to read the graphs appropriately
em_abundance_borehole3 <- read.csv("AbundanceLinesB3.csv")
em_abundance_borehole3$dates <- as.Date(em_abundance_borehole3$dates)

palette_borehole2 <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7") #Color palette for Borehole 2
palette_borehole3 <- c("#E69F00", "#56B4E9", "#009E73", "#0072B2", "#CC79A7", "#D55E00") #Color palette for Borehole 3
```

### Creating Abundance Lines Figure for Borehole 2
```{r}
borehole2_abundance <- ggplot(em_abundance_borehole2, aes_string(x = "dates", y = "Abundance", 
                                     group = "Phylum"))

borehole2_abundance <- borehole2_abundance + geom_point(size = 1.5, aes_string(color = "Phylum"), 
                        show.legend = FALSE)

borehole2_abundance <- borehole2_abundance + geom_line(size = 1.2, aes_string(color = "Phylum")) + 
    facet_grid(reformulate("Site_Collected"), scales = "free", 
               space = "free") + scale_colour_manual(values = palette_borehole2) + 
    guides(colour = guide_legend(ncol = ceiling(length(unique(em_abundance_borehole2[["Phylum"]]))/25), 
                                 override.aes = list(size = 4)))

borehole2_abundance <- borehole2_abundance + ylab("Relative Abundance")

borehole2_abundance <- borehole2_abundance + theme_bw() + theme(axis.text.x = element_text(angle = 30, 
                                                         hjust = 1, size = 10), axis.text.y = element_text(hjust = 0.95, 
                                                                                                           size = 10), axis.title.x = element_text(size = 10, face = "bold"), 
                              axis.title.y = element_text(size = 10, face = "bold"), 
                              axis.ticks.x = element_blank(), legend.title = element_text(size = 12, 
                                                                                          face = "bold"), legend.text = element_text(size = 8), 
                              legend.spacing.x = unit(0.005, "npc"), legend.key.size = unit(4, 
                                                                                            "mm"), legend.background = element_rect(fill = (alpha = 0)), 
                              panel.background = element_rect(color = "black", size = 2, 
                                                              fill = "white"), panel.spacing = unit(0.015, "npc"), 
                              strip.text.x = element_text(size = 10, face = "bold", 
                                                          color = "black"), strip.background = element_rect(colour = "black", 
                                                                                                            size = 1.4, fill = "white")) + scale_y_continuous(expand = expansion(mult = c(0, 
                                                                                                                                                                                          0.003), add = c(0.0015, 0.001))) + scale_x_discrete(expand = expansion(mult = 0, add = c(0.3, 0.5)))

borehole2_abundance + scale_x_date(labels = label_date(format = "%Y-%m"), breaks = breaks_width("4 months")) +
  geom_rect(aes(xmin = ymd('2017-09-02'), xmax = ymd('2017-12-01'), ymin = -Inf, ymax = Inf), fill = "grey", alpha = 0.01) + #Indicating Fall seasonal collection
  geom_rect(aes(xmin = ymd('2018-09-02'), xmax = ymd('2018-12-01'), ymin = -Inf, ymax = Inf), fill = "grey", alpha = 0.01) + #Indicating Fall seasonal collection
  geom_rect(aes(xmin = ymd('2019-06-02'), xmax = ymd('2019-09-01'), ymin = -Inf, ymax = Inf), fill = "#000000", alpha = 0.01) + #Indicating Summer seasonal collection
  geom_rect(aes(xmin = ymd('2020-06-02'), xmax = ymd('2020-09-01'), ymin = -Inf, ymax = Inf), fill = "#000000", alpha = 0.01) + #Indicating Summer seasonal collection
  labs(x = "") + 
  grids(linetype = "dashed")
```


### Creating Abundance Lines Figure for Borehole 3
```{r}
borehole3_abundance <- ggplot(em_abundance_borehole3, aes_string(x = "dates", y = "Abundance", 
                                     group = "Phylum"))

borehole3_abundance <- borehole3_abundance + geom_point(size = 1.5, aes_string(color = "Phylum"), 
                        show.legend = FALSE)

borehole3_abundance <- borehole3_abundance + geom_line(size = 1.2, aes_string(color = "Phylum")) + 
    facet_grid(reformulate("Site_Collected"), scales = "free", 
               space = "free") + scale_colour_manual(values = palette_borehole3) + 
    guides(colour = guide_legend(ncol = ceiling(length(unique(em_abundance_borehole3[["Phylum"]]))/25), 
                                 override.aes = list(size = 4)))

borehole3_abundance <- borehole3_abundance + ylab("Relative Abundance")

borehole3_abundance <- borehole3_abundance + theme_bw() + theme(axis.text.x = element_text(angle = 30, 
                                                         hjust = 1, size = 10), axis.text.y = element_text(hjust = 0.95, 
                                                                                                           size = 10), axis.title.x = element_text(size = 10, face = "bold"), 
                              axis.title.y = element_text(size = 10, face = "bold"), 
                              axis.ticks.x = element_blank(), legend.title = element_text(size = 12, 
                                                                                          face = "bold"), legend.text = element_text(size = 8), 
                              legend.spacing.x = unit(0.005, "npc"), legend.key.size = unit(4, 
                                                                                            "mm"), legend.background = element_rect(fill = (alpha = 0)), 
                              panel.background = element_rect(color = "black", size = 2, 
                                                              fill = "white"), panel.spacing = unit(0.015, "npc"), 
                              strip.text.x = element_text(size = 10, face = "bold", 
                                                          color = "black"), strip.background = element_rect(colour = "black", 
                                                                                                            size = 1.4, fill = "white")) + scale_y_continuous(expand = expansion(mult = c(0, 
                                                                                                                                                                                          0.003), add = c(0.0015, 0.001))) + scale_x_discrete(expand = expansion(mult = 0, add = c(0.3, 0.5)))

borehole3_abundance + scale_x_date(labels = label_date(format = "%Y-%m"), breaks = breaks_width("4 months")) +
  geom_rect(aes(xmin = ymd('2019-06-02'), xmax = ymd('2019-09-01'), ymin = -Inf, ymax = Inf), fill = "#000000", alpha = 0.01) + #Indicating Summer seasonal collection
  geom_rect(aes(xmin = ymd('2020-06-02'), xmax = ymd('2020-09-01'), ymin = -Inf, ymax = Inf), fill = "#000000", alpha = 0.01) + #Indicating Summer seasonal collection
  geom_rect(aes(xmin = ymd('2018-09-02'), xmax = ymd('2018-12-01'), ymin = -Inf, ymax = Inf), fill = "grey", alpha = 0.01) + #Indicating Fall seasonal collection
  labs(x = "") + 
  grids(linetype = "dashed")
```


# Geochemistry Data

### Loading the Appropriate datasets containing the chemical and isotopic data from the study
```{r}
em_h20 <- read.csv("dH20_data.csv", header = TRUE)

em_meta_lm <- read.csv("geochem_data.csv", row.names = 1) #The geochemistry data is reduced from the completed published data in the manuscript to include the analytes of interest from the study
```


### Creating Mn-Zn Linear Regression Figure
```{r}
em_linear_reg <- em_meta_lm %>%
  ggplot(aes(x = Mn.257.610...A, y = Zn.213.857...A)) +
  geom_point(mapping = aes(color = Site.Collected.From), size = 6) +
  geom_smooth(method = "lm") +
  labs(x = "Mn (ppm)", y = "Zn (ppm)", color = "") +
  scale_color_manual(values = c("#0072B2", "#F0E442", "#E69F00","#009E73")) +
  theme_bw() +
  theme(axis.title.x = element_text(size = 16, face = "bold"), axis.title.y = element_text(size = 16, face = "bold"), axis.text.x = element_text(size = 12), axis.text.y = element_text(size = 12), legend.text = element_text(size = 15))
em_linear_reg
```


#### Calculating P-value, Adjusted R-Squared value, and F-statistic
```{r}
Mn_Zn_lm <- lm(Mn.257.610...A ~ Zn.213.857...A, data = em_meta_lm)
summary(Mn_Zn_lm)
```


### Creating Water-Isotope Figure

```{r}
em_H2O_isotope_plot <- ggplot(data = em_h20) +
  geom_point(mapping = aes(x = d_18O, y = d_2H, shape = Borehole, fill = Borehole), size = 6) +
  geom_abline(slope = 8, intercept = 10) +
  geom_abline(slope = 8, intercept = 15, linetype = 2) +
  geom_abline(slope = 8, intercept = 5, linetype = 2, color = "black") +
  scale_shape_manual(values = c(21,22,24)) +
  scale_fill_manual(values = c("#0072B2", "#F0E442", "#009E73")) + 
  xlab(expression(paste(delta^{18}, "O(\u2030) VSMOW"))) + 
  ylab(expression(paste(delta^{2}, "H(\u2030) VSMOW"))) +
  labs(shape = "", fill = "") +
  xlim(-17, -16.5) +
  ylim(-140, -110) +
  theme_classic() +
  theme(axis.title = element_text(size = 20), axis.text = element_text(size = 15), legend.text = element_text(size = 15))
em_H2O_isotope_plot
```
